### DISCUSSION TITLE
how to recognize scanned pdf quickly?

### DISCUSSION THREAD

**Discussion by GodsDusk** [jsvine/pdfplumber#688] [Ask for help with specific PDFs]
I have an 8M scanned pdf with 2 pages, and use

```python
with pdfplumber.open(file) as pdf:
    if not pdf.chars:
        return
```
to recognize scanned pdf.

but this code cost almost 10 mins on my computer(ubuntu, i7-5930K)

how can I improve this?

**Comment by jsvine** [comment]
Hi @GodsDusk, and thanks for your interest in this library. I agree: It'd be great to try speeding things up. Are you able to share the PDF? That'd certainly help to diagnose the slowness.

**Reply by GodsDusk** [reply]
[78、利伐沙班片+正大天晴药业集团股份有限公司.pdf](https://github.com/jsvine/pdfplumber/files/9139284/78.%2B.pdf)

here is the file

**Reply by jsvine** [reply]
Thanks! Seeing the PDF helped a lot. It turns out that this is *not* a scanned PDF, but rather a digital PDF with a *lot* of graphical elements in it — the characters aren't standard PDF text, but rather drawn through ~47,000 PDF paths (lines, curves, rects). Here's a demonstration: https://notebooksharing.space/view/c09916f1f6180a8d074461ea30e714cd87db84aa9eb27e83ec44234071c1fb8b

Unfortunately, this means it will be difficult to speed things up without speeding up `pdfminer.six`'s parser. (That's this library's main dependency.)

**Reply by GodsDusk** [reply]
pdf is evil ! 😂
thanks a lot
